src/main/java/javax/validation/spi/ConfigurationState.java

```diff
@@ -81,6 +82,31 @@
 	Set<InputStream> getMappingStreams();
 
 	/**
+	 * Returns a set of value extractors.
+	 * <p>
+	 * The extractors are retrieved from the following sources in decreasing
+	 * order:
+	 * <ul>
+	 *     <li>extractors passed programmatically to {@link Configuration}</li>
+	 *     <li>extractors defined in {@code META-INF/validation.xml} provided
+	 *     that {@code ignoredXmlConfiguration} is {@code false}.</li>
+	 *     <li>extractors loaded through the Java service loader</li>
+	 * </ul>
+	 * An extractor for a given type and type parameter passed in
+	 * programmatically takes precedence over any extractors for the same type
+	 * and type parameter defined in {@code META-INF/validation.xml} or loaded
+	 * through the service loader. Extractors defined in
+	 * {@code META-INF/validation.xml} take precedence over any extractor for
+	 * the same type and type parameter loaded through the service loader.
+	 * <p>
+	 * Extractors defined in {@code META-INF/validation.xml} or loaded through
+	 * the service loader are instantiated using their no-arg constructor.
```

Comment:
It's done for other extension points, too, e.g. see `getConstraintValidatorFactory()`.